pkg/controllers/provisioning/scheduling/suite_test.go

```diff
@@ -133,6 +133,7 @@ var _ = Context("Scheduling", func() {
 								},
 							},
 						},
+						StartupTaints: []v1.Taint{v1beta1.UnregisteredNoExecuteTaint},
```

Comment:
I don't think that we should have to manually add this as a startup taint on the NodePool; however, we should make sure that we properly consider the taints correctly using the cluster state node before the node has been marked as registered

Reply:
Done.